azure/services/scalesetvms/scalesetvms.go

```diff
@@ -92,7 +90,7 @@ func (s *Service) Delete(ctx context.Context) error {
 		instanceID    = s.Scope.InstanceID()
 	)
 
-	log := s.Scope.WithValues("resourceGroup", resourceGroup, "scaleset", vmssName, "instanceID", instanceID)
+	log = log.WithValues("resourceGroup", resourceGroup, "scaleset", vmssName, "instanceID", instanceID)
```

Comment:
can we not plug this in `StartSpanWithLogger `? or is this intentionally done to not include the kvs in spans?